elements/lisk-codec/src/codec.ts

```diff
@@ -22,20 +24,31 @@ import {
 	Schema,
 	SchemaProps,
 } from './types';
-
-import { writeObject, readObject } from './collection';
+import { validateSchema } from './schema';
+import { SchemaError } from './schema/errors';
 
 export class Codec {
 	private readonly _compileSchemas: CompiledSchemas = {};
 
-	public addSchema(schema: Schema): void {
+	public addSchema(schema: Schema): ReadonlyArray<ErrorObject> {
+		const errors = validateSchema(schema);
+		if (errors.length) {
+			return errors;
+		}
+
 		const schemaName = schema.$id;
 		this._compileSchemas[schemaName] = this._compileSchema(schema, [], []);
+
+		return [];
```

Comment:
Shouldn't we just throw error here? in encode and decode it is throwing an error

Reply:
We can do that as well. I tried to keep that interface compliant with other validation codes where we return errors.